src/instruction/incl.rs

```diff
@@ -0,0 +1,184 @@
+//! This module is used to parse external code and make it available to other source
+//! files.
+
+use std::path::{Path, PathBuf};
+
+use crate::{parser::Construct, InstrKind, Instruction, Interpreter, JkErrKind, JkError};
+
+/// An `Incl` is constituted of a path, an optional alias and contains an interpreter.
+/// The interpreter is built from parsing the source file in the path.
+/// Aliases are used to potentially rename exported functions.
+#[derive(Clone)]
+pub struct Incl {
+    path: String,
+    alias: Option<String>,
+}
+
+/// Default file that gets included when including a directory in jinko source code
+const DEFAULT_INCL: &str = "/lib.jk";
+
+impl Incl {
+    pub fn new(path: String, alias: Option<String>) -> Incl {
+        Incl { path, alias }
+    }
+
+    /// Rename all contained code to the correct alias
+    fn _rename(&mut self) {
+        todo!("Implement once namespaces are implemented")
+    }
+
+    fn format_candidates(&self, base: &Path) -> (PathBuf, PathBuf) {
+        let mut format = PathBuf::from(base);
+        format.push(&self.path);
+
+        // FIXME: No unwrap
+        let format = format.to_str().unwrap().to_string();
+
+        let mut dir_fmt = format.clone();
+        let mut file_fmt = format.clone();
+
+        dir_fmt.push_str(&format!("{}", DEFAULT_INCL));
+        file_fmt.push_str(".jk");
+
+        (PathBuf::from(dir_fmt), PathBuf::from(file_fmt))
+    }
+
+    fn format_path(&self, base: &Path) -> Result<PathBuf, JkError> {
```

Comment:
I'm not sure about the name. If the goal is to find the file to include, why not ``find_include_path`` or something along those lines ?

Reply:
Done